src/test/java/io/strimzi/test/container/StrimziProducerConsumerContainerKRaftIT.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.test.container;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.utility.MountableFile;
+
+import java.time.Duration;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class StrimziProducerConsumerContainerKRaftIT extends AbstractIT {
+
+    private static StrimziKafkaContainer strimziKafkaContainer;
+    private static String bootstrapServers;
+
+    @BeforeAll
+    static void setup() {
+        strimziKafkaContainer = new StrimziKafkaContainer()
+            .withKraft()
+            .withKafkaConfigurationMap(Map.of("auto.create.topics.enable", "true"))
+            .waitForRunning();
+
+        strimziKafkaContainer.start();
+        bootstrapServers = strimziKafkaContainer.getInternalBootstrapServers();
+    }
+
+    @AfterAll
+    static void teardown() {
+        if (strimziKafkaContainer != null) {
+            strimziKafkaContainer.stop();
+        }
+    }
+
+    @Test
+    void testProducerAndConsumerWithDefaultConfigs() {
+        String topicName = "default-topic";
+        String messageContent = "Test message for default configs\n";
+
+        // Start Producer
+        StrimziProducerContainer producerContainer = new StrimziProducerContainer()
+            .withBootstrapServer(bootstrapServers)
+            .withTopic(topicName)
+            .withMessageContent(messageContent)
+            .withSync()
+            .withLogging();
+
+        producerContainer.start();
+
+        // Start Consumer
+        StrimziConsumerContainer consumerContainer = new StrimziConsumerContainer()
+            .withBootstrapServer(bootstrapServers)
+            .withTopic(topicName)
+            .withGroup("default-group")
+            .withFromBeginning()
+            .withLogging();
+
+        consumerContainer.start();
+
+        // Wait for the consumer to receive the message
+        Utils.waitFor("Consumer receives message", Duration.ofSeconds(1).toMillis(), Duration.ofSeconds(20).toMillis(),
```

Comment:
Not directly related to this PR but `Utils.waitFor()` should accept `Duration` arguments and convert those into the unit it needs internally. Otherwise calling it is really cumbersome. I find `Duration.ofSeconds(1).toMillis()` less readable than `1000`.